python/intro/README.md

```diff
@@ -0,0 +1,405 @@
+    name = "World"
+    if data and len(data) > 0:
+        body = json.loads(data)
+        name = body.get("name")
+    return {"message": "Hello {0}".format(name)}
+
+
+
+if __name__ == "__main__":
+    fdk.handle(handler)
+
+```
+
+This function looks for JSON input in the form of `{"name": "Bob"}`. If this
+JSON example is passed to the function, the function returns `{"message":"Hello
+Bob!"}`. If no JSON data is found, the function returns `{"message":"Hello
+World!"}`.  
+
+### Understanding func.yaml
+The `fn init` command generated a `func.yaml` function
+configuration file. Let's look at the contents:
+
+![user input](images/userinput.png)
+>```sh
+> cat func.yaml
+>```
+
+```yaml
+name: pythonfn
+version: 0.0.1
+runtime: python
+entrypoint: python3 func.py
+format: json
+```
+
+The generated `func.yaml` file contains metadata about your function and
+declares a number of properties including:
+
+* name--the name of the function. Matches the directory name.
+* version--automatically starting at 0.0.1.
+* runtime--the name of the runtime/language which was set based on the value set
+in `--runtime`.
+* entrypoint--the name of the executable to invoke when your function is called,
+in this case `python3 func.rb`.
+* format--the function uses JSON as its input/output method ([see: Open Function Format](https://github.com/fnproject/fn/blob/master/docs/developers/function-format.md)).
+
+There are other user specifiable properties but these will suffice for
+this example.  Note that the name of your function is taken from the containing
+folder name.  We'll see this come into play later on.
+
+### Other Function Files
+The `fn init` command generated two other files.
+
+* `requirements.txt` --  specifies all the dependencies for your Python
+function.
+* `test.json` -- a test file that is used to test your function. It defines an
+input and the output of the function and helps to identify if the function works
+correctly or not. Function testing is not covered in this tutorial.
+
+
+## Running Your First Function
+With the `pythonfn` directory containing `func.py` and `func.yaml` you've
+got everything you need to run the function.  So let's run it and
+observe the output.  Note that the first time you build a
+function of a particular language it takes longer as Fn downloads
+the necessary Docker images.
+
+![user input](images/userinput.png)
+>```sh
+> fn run
+>```
+
+```yaml
+Building image fndemouser/pythonfn:0.0.1
+{"message":"Hello World"}
+```
+
+The last line of output is `{"message":"Hello World"}` since no input was passed
+to the function.
+
+If you ever want more details on what a given fn command is doing behind the
+scenes you can add the `--verbose` switch.  Let's rerun with verbose output
+enabled.
+
+![user input](images/userinput.png)
+>```sh
+> fn --verbose run
+>```
+
+```yaml
+Building image pythonfn:0.0.1
+Sending build context to Docker daemon  6.144kB
+Step 1/6 : FROM python:3.6
```

Comment:
you need to update your CLI, because python wrapper uses different python image (smaller image: `python:3.6-slim-stretch`)

Reply:
Fixed.